src/app/firebase.service.ts

```diff
@@ -66,6 +66,21 @@ export class FirebaseService {
       .valueChanges();
   }
 
+  /**
+   * Adds round to firestore and sets round to the session's current round
+   * @param sessionID Session ID to which round will be added
+   * @param roundData data for the round in FirebaseRound interface format
+   */
+  public async createRoundInSession(sessionID: string, roundData: FirebaseRound){
+    let round: Promise<DocumentReference>;
+    round = this.firestore.collection('sessions/' + sessionID + '/rounds').add(roundData);
+    this.setCurrentRound({sessionId: sessionID, roundId: (await round).id});
+  }
```

Comment:
This logic should be moved to the session service. The Firebase service should only contain the most basic wrappers around Firebase functionality, so in this case `this.firestore.collection('sessions/' + sessionID + '/rounds').add(roundData);` would be abstracted into its own function in the Firebase service and `createRoundInSession` should be moved to the teacher session (or maybe round) service and get the session from the current session subject. Additionally, `createRoundInSession` should return a promise of the result of setting the current round so that it can be waited for.

Reply:
Oh, even better, yeah!